ci_cd/tasks/update_deps.py

```diff
@@ -181,13 +188,24 @@ def update_deps(  # pylint: disable=too-many-branches,too-many-locals,too-many-s
         dependencies.extend(optional_deps)
 
     # Placeholder and default variables
-    updated_packages = {}
-    error = False
+    already_handled_packages: set[Requirement] = set()
+    updated_packages: dict[str, str] = {}
+    error: bool = False
 
     for dependency in dependencies:
         try:
             parsed_requirement = Requirement(dependency)
         except InvalidRequirement as exc:
+            if skip_unnormalized_python_package_names:
+                msg = (
+                    f"Skipping requirement {dependency!r}, as unnormalized Python "
+                    "package naming is allowed by user. Note, the requirements could "
+                    f"not be parsed: {exc}"
+                )
+                LOGGER.info(msg)
```

Comment:
I'm not sure if I like the fact that both LOGGER and print are being used to pass messages to the user